lib/Config.js

```diff
@@ -729,6 +729,13 @@ class Config extends EventEmitter {
         this.authData.accounts = accounts;
         this.emit('authdata-update');
     }
+
+    getAwsBucketName(locationConstraint) {
+        return this.locationConstraints[locationConstraint].details.bucketName;
+    }
+    isBucketMatch(locationConstraint) {
+        return this.locationConstraints[locationConstraint].details.bucketMatch;
```

Comment:
this function and the one above could throw if the locationConstraint is unknown or doesn't have a details section. are we ok with that? should we have a check first and return undefined for the bucketname and the bucketMatch if not known?

Reply:
`locationConstraint` comes from the locationConstraint metadata of the source object.
Before storing the object source locationConstraint, we check the object location constraint so we don't save an unknown location.
here: lib/api/apiUtils/object/locationConstraintCheck.js
Also, in our `Config.json` we make sure that `locationConstraints[l].details` is an object
So I don't think we need to check that the location is known and if `bucketName` or `BucketMatch` are not specified in our `locationConfig.json` it will be undefined (no error will be thrown) .

Reply:
well, if we have corrupted metadata we could throw but that would be a good wake up call to a problem.